bftengine/src/bftengine/messages/AskForCheckpointMsg.hpp

```diff
@@ -13,27 +13,37 @@
 
 #include "MessageBase.hpp"
 #include "Digest.hpp"
+#include "assertUtils.hpp"
 
 namespace bftEngine::impl {
 
 class AskForCheckpointMsg : public MessageBase {
  public:
-  static MsgSize maxSizeOfAskForCheckpointMsg() { return sizeof(AskForCheckpointMsgHeader); }
+  static MsgSize maxSizeOfAskForCheckpointMsg() { return sizeof(AskForCheckpointMsgHeader) + SPAN_CONTEXT_MAX_SIZE; }
 
   static MsgSize maxSizeOfAskForCheckpointMsgInLocalBuffer() {
     return maxSizeOfAskForCheckpointMsg() + sizeof(RawHeaderOfObjAndMsg);
   }
 
-  AskForCheckpointMsg(ReplicaId senderId)
-      : MessageBase(senderId, MsgCode::AskForCheckpoint, sizeof(AskForCheckpointMsgHeader)) {}
+  AskForCheckpointMsg(ReplicaId senderId, const std::string& spanContext = "")
+      : MessageBase(senderId, MsgCode::AskForCheckpoint, spanContext.size(), sizeof(AskForCheckpointMsgHeader)) {
+    char* position = body() + sizeof(AskForCheckpointMsgHeader);
+    memcpy(position, spanContext.data(), spanContext.size());
+  }
+
+  std::string spanContext() const override {
+    return std::string(body() + sizeof(AskForCheckpointMsgHeader), spanContextSize());
+  }
 
   AskForCheckpointMsg* clone() { return new AskForCheckpointMsg(*this); }
 
   void validate(const ReplicasInfo& repInfo) const override {
     Assert(type() == MsgCode::AskForCheckpoint);
     Assert(senderId() != repInfo.myId());
 
-    if (size() > sizeof(AskForCheckpointMsgHeader)) throw std::runtime_error(__PRETTY_FUNCTION__);
+    if (size() > sizeof(AskForCheckpointMsgHeader) + spanContextSize()) {
```

Comment:
I support the idea to have a base class method to check the validity of the messages.
However, since the validity conditions vary from message to message, it requires a deeper analysis.
Since PR is already quite large, I'd prefer to have a separate PR for this change.
WDYT?

Reply:
Makes sense to me. Fine to leave as is for now.